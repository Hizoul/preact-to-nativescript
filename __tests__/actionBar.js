import * as preactToNativeScript from "../index"

const h = preactToNativeScript.Preact.h

test("Action Bar", () => {
  expect(true).toBe(true)
  const resultingRender = preactToNativeScript.render(
    h("page", null, [
      h("actionBar", null, [
        h("stackLayout", null, [
          h("label", {text: "myText"})
        ]),
        h("navigationButton", {text: "MyNav"}, [h("label", {}, ["rawtext", "as", "children"])]),
        h("actionItem", {text: "myAction"}),
        h("actionItem", {text: "otherAction"}, [h("label", {text: "ot"})])
      ]),
      h("scrollView", null, [
        h("stackLayout", null, [
          h("label", {text: "myText"})
        ])
      ])
    ])
  )
  expect(resultingRender).toMatchSnapshot("Render of ActionBar")
})
