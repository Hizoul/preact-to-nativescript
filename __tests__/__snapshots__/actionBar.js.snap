// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Bar: Render of ActionBar 1`] = `
Page {
  "__n": "page",
  "__preactattr_": Object {},
  "actionBar": ActionBar {
    "__n": "actionBar",
    "__preactattr_": Object {},
    "actionItems": Object {
      "addItem": [Function],
      "removeItem": [Function],
    },
    "actionItemsMock": Array [
      ActionItem {
        "__n": "actionItem",
        "__preactattr_": Object {
          "text": "myAction",
        },
        "attributes": Array [],
        "childNodes": Array [],
        "firstLoad": true,
        "loaded": false,
        "nodeName": "ACTIONITEM",
        "nodeType": 1,
        "parentNode": [Circular],
        "set": [Function],
        "text": "myAction",
      },
      ActionItem {
        "__n": "actionItem",
        "__preactattr_": Object {
          "text": "otherAction",
        },
        "attributes": Array [],
        "childNodes": Array [],
        "firstLoad": true,
        "loaded": false,
        "nodeName": "ACTIONITEM",
        "nodeType": 1,
        "parentNode": [Circular],
        "set": [Function],
        "text": "otherAction",
      },
    ],
    "attributes": Array [],
    "childNodes": Array [
      StackLayout {
        "__n": "stackLayout",
        "__preactattr_": Object {},
        "attributes": Array [],
        "childNodes": Array [
          Label {
            "__n": "label",
            "__preactattr_": Object {
              "text": "myText",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "LABEL",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "text": "myText",
          },
        ],
        "firstLoad": true,
        "loaded": false,
        "nodeName": "STACKLAYOUT",
        "nodeType": 1,
        "parentNode": [Circular],
        "set": [Function],
      },
      NavigationButton {
        "__n": "navigationButton",
        "__preactattr_": Object {
          "text": "MyNav",
        },
        "attributes": Array [],
        "childNodes": Array [],
        "firstLoad": true,
        "loaded": false,
        "nodeName": "NAVIGATIONBUTTON",
        "nodeType": 1,
        "parentNode": [Circular],
        "set": [Function],
        "text": "MyNav",
      },
      ActionItem {
        "__n": "actionItem",
        "__preactattr_": Object {
          "text": "myAction",
        },
        "attributes": Array [],
        "childNodes": Array [],
        "firstLoad": true,
        "loaded": false,
        "nodeName": "ACTIONITEM",
        "nodeType": 1,
        "parentNode": [Circular],
        "set": [Function],
        "text": "myAction",
      },
      ActionItem {
        "__n": "actionItem",
        "__preactattr_": Object {
          "text": "otherAction",
        },
        "attributes": Array [],
        "childNodes": Array [],
        "firstLoad": true,
        "loaded": false,
        "nodeName": "ACTIONITEM",
        "nodeType": 1,
        "parentNode": [Circular],
        "set": [Function],
        "text": "otherAction",
      },
    ],
    "firstLoad": true,
    "loaded": false,
    "navigationButton": NavigationButton {
      "__n": "navigationButton",
      "__preactattr_": Object {
        "text": "MyNav",
      },
      "attributes": Array [],
      "childNodes": Array [],
      "firstLoad": true,
      "loaded": false,
      "nodeName": "NAVIGATIONBUTTON",
      "nodeType": 1,
      "parentNode": [Circular],
      "set": [Function],
      "text": "MyNav",
    },
    "nodeName": "ACTIONBAR",
    "nodeType": 1,
    "parentNode": [Circular],
    "set": [Function],
    "titleView": StackLayout {
      "__n": "stackLayout",
      "__preactattr_": Object {},
      "attributes": Array [],
      "childNodes": Array [
        Label {
          "__n": "label",
          "__preactattr_": Object {
            "text": "myText",
          },
          "attributes": Array [],
          "childNodes": Array [],
          "firstLoad": true,
          "loaded": false,
          "nodeName": "LABEL",
          "nodeType": 1,
          "parentNode": [Circular],
          "set": [Function],
          "text": "myText",
        },
      ],
      "firstLoad": true,
      "loaded": false,
      "nodeName": "STACKLAYOUT",
      "nodeType": 1,
      "parentNode": [Circular],
      "set": [Function],
    },
  },
  "attributes": Array [],
  "childNodes": Array [
    ActionBar {
      "__n": "actionBar",
      "__preactattr_": Object {},
      "actionItems": Object {
        "addItem": [Function],
        "removeItem": [Function],
      },
      "actionItemsMock": Array [
        ActionItem {
          "__n": "actionItem",
          "__preactattr_": Object {
            "text": "myAction",
          },
          "attributes": Array [],
          "childNodes": Array [],
          "firstLoad": true,
          "loaded": false,
          "nodeName": "ACTIONITEM",
          "nodeType": 1,
          "parentNode": [Circular],
          "set": [Function],
          "text": "myAction",
        },
        ActionItem {
          "__n": "actionItem",
          "__preactattr_": Object {
            "text": "otherAction",
          },
          "attributes": Array [],
          "childNodes": Array [],
          "firstLoad": true,
          "loaded": false,
          "nodeName": "ACTIONITEM",
          "nodeType": 1,
          "parentNode": [Circular],
          "set": [Function],
          "text": "otherAction",
        },
      ],
      "attributes": Array [],
      "childNodes": Array [
        StackLayout {
          "__n": "stackLayout",
          "__preactattr_": Object {},
          "attributes": Array [],
          "childNodes": Array [
            Label {
              "__n": "label",
              "__preactattr_": Object {
                "text": "myText",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "LABEL",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "text": "myText",
            },
          ],
          "firstLoad": true,
          "loaded": false,
          "nodeName": "STACKLAYOUT",
          "nodeType": 1,
          "parentNode": [Circular],
          "set": [Function],
        },
        NavigationButton {
          "__n": "navigationButton",
          "__preactattr_": Object {
            "text": "MyNav",
          },
          "attributes": Array [],
          "childNodes": Array [],
          "firstLoad": true,
          "loaded": false,
          "nodeName": "NAVIGATIONBUTTON",
          "nodeType": 1,
          "parentNode": [Circular],
          "set": [Function],
          "text": "MyNav",
        },
        ActionItem {
          "__n": "actionItem",
          "__preactattr_": Object {
            "text": "myAction",
          },
          "attributes": Array [],
          "childNodes": Array [],
          "firstLoad": true,
          "loaded": false,
          "nodeName": "ACTIONITEM",
          "nodeType": 1,
          "parentNode": [Circular],
          "set": [Function],
          "text": "myAction",
        },
        ActionItem {
          "__n": "actionItem",
          "__preactattr_": Object {
            "text": "otherAction",
          },
          "attributes": Array [],
          "childNodes": Array [],
          "firstLoad": true,
          "loaded": false,
          "nodeName": "ACTIONITEM",
          "nodeType": 1,
          "parentNode": [Circular],
          "set": [Function],
          "text": "otherAction",
        },
      ],
      "firstLoad": true,
      "loaded": false,
      "navigationButton": NavigationButton {
        "__n": "navigationButton",
        "__preactattr_": Object {
          "text": "MyNav",
        },
        "attributes": Array [],
        "childNodes": Array [],
        "firstLoad": true,
        "loaded": false,
        "nodeName": "NAVIGATIONBUTTON",
        "nodeType": 1,
        "parentNode": [Circular],
        "set": [Function],
        "text": "MyNav",
      },
      "nodeName": "ACTIONBAR",
      "nodeType": 1,
      "parentNode": [Circular],
      "set": [Function],
      "titleView": StackLayout {
        "__n": "stackLayout",
        "__preactattr_": Object {},
        "attributes": Array [],
        "childNodes": Array [
          Label {
            "__n": "label",
            "__preactattr_": Object {
              "text": "myText",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "LABEL",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "text": "myText",
          },
        ],
        "firstLoad": true,
        "loaded": false,
        "nodeName": "STACKLAYOUT",
        "nodeType": 1,
        "parentNode": [Circular],
        "set": [Function],
      },
    },
    ScrollView {
      "__n": "scrollView",
      "__preactattr_": Object {},
      "attributes": Array [],
      "childNodes": Array [
        StackLayout {
          "__n": "stackLayout",
          "__preactattr_": Object {},
          "attributes": Array [],
          "childNodes": Array [
            Label {
              "__n": "label",
              "__preactattr_": Object {
                "text": "myText",
              },
              "attributes": Array [],
              "childNodes": Array [],
              "firstLoad": true,
              "loaded": false,
              "nodeName": "LABEL",
              "nodeType": 1,
              "parentNode": [Circular],
              "set": [Function],
              "text": "myText",
            },
          ],
          "firstLoad": true,
          "loaded": false,
          "nodeName": "STACKLAYOUT",
          "nodeType": 1,
          "parentNode": [Circular],
          "set": [Function],
        },
      ],
      "content": StackLayout {
        "__n": "stackLayout",
        "__preactattr_": Object {},
        "attributes": Array [],
        "childNodes": Array [
          Label {
            "__n": "label",
            "__preactattr_": Object {
              "text": "myText",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "LABEL",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "text": "myText",
          },
        ],
        "firstLoad": true,
        "loaded": false,
        "nodeName": "STACKLAYOUT",
        "nodeType": 1,
        "parentNode": [Circular],
        "set": [Function],
      },
      "firstLoad": true,
      "loaded": false,
      "nodeName": "SCROLLVIEW",
      "nodeType": 1,
      "parentNode": [Circular],
      "set": [Function],
    },
  ],
  "content": ScrollView {
    "__n": "scrollView",
    "__preactattr_": Object {},
    "attributes": Array [],
    "childNodes": Array [
      StackLayout {
        "__n": "stackLayout",
        "__preactattr_": Object {},
        "attributes": Array [],
        "childNodes": Array [
          Label {
            "__n": "label",
            "__preactattr_": Object {
              "text": "myText",
            },
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "LABEL",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "text": "myText",
          },
        ],
        "firstLoad": true,
        "loaded": false,
        "nodeName": "STACKLAYOUT",
        "nodeType": 1,
        "parentNode": [Circular],
        "set": [Function],
      },
    ],
    "content": StackLayout {
      "__n": "stackLayout",
      "__preactattr_": Object {},
      "attributes": Array [],
      "childNodes": Array [
        Label {
          "__n": "label",
          "__preactattr_": Object {
            "text": "myText",
          },
          "attributes": Array [],
          "childNodes": Array [],
          "firstLoad": true,
          "loaded": false,
          "nodeName": "LABEL",
          "nodeType": 1,
          "parentNode": [Circular],
          "set": [Function],
          "text": "myText",
        },
      ],
      "firstLoad": true,
      "loaded": false,
      "nodeName": "STACKLAYOUT",
      "nodeType": 1,
      "parentNode": [Circular],
      "set": [Function],
    },
    "firstLoad": true,
    "loaded": false,
    "nodeName": "SCROLLVIEW",
    "nodeType": 1,
    "parentNode": [Circular],
    "set": [Function],
  },
  "firstLoad": true,
  "loaded": false,
  "mergeInto": undefined,
  "nodeName": "PAGE",
  "nodeType": 1,
  "renderedComponent": VNode {
    "attributes": undefined,
    "children": Array [
      VNode {
        "attributes": undefined,
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              VNode {
                "attributes": Object {
                  "text": "myText",
                },
                "children": Array [],
                "key": undefined,
                "nodeName": "label",
              },
            ],
            "key": undefined,
            "nodeName": "stackLayout",
          },
          VNode {
            "attributes": Object {
              "text": "MyNav",
            },
            "children": Array [],
            "key": undefined,
            "nodeName": "navigationButton",
          },
          VNode {
            "attributes": Object {
              "text": "myAction",
            },
            "children": Array [],
            "key": undefined,
            "nodeName": "actionItem",
          },
          VNode {
            "attributes": Object {
              "text": "otherAction",
            },
            "children": Array [],
            "key": undefined,
            "nodeName": "actionItem",
          },
        ],
        "key": undefined,
        "nodeName": "actionBar",
      },
      VNode {
        "attributes": undefined,
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              VNode {
                "attributes": Object {
                  "text": "myText",
                },
                "children": Array [],
                "key": undefined,
                "nodeName": "label",
              },
            ],
            "key": undefined,
            "nodeName": "stackLayout",
          },
        ],
        "key": undefined,
        "nodeName": "scrollView",
      },
    ],
    "key": undefined,
    "nodeName": "page",
  },
  "renderedInto": [Circular],
  "set": [Function],
}
`;
